src/CppAst/CppModelBuilder.cs

```diff
@@ -2024,118 +2026,118 @@ protected void WarningUnhandled(CXCursor cursor, CXCursor parent)
-                        var cppClass = new CppClass(GetCursorSpelling(templateCursor));
-                        cppClass.IsAnonymous = templateCursor.IsAnonymous;
-                        cppClass.IsAbstract = templateCursor.CXXRecord_IsAbstract;
-                        cppClass.TemplateKind = CppTemplateKind.TemplateClass;
-                        templateCursor.VisitChildren((childCursor, classCursor, clientData) =>
-                        {
-                            var tmplParam = ParseTemplateParameters(childCursor, clientData);
-                            if (tmplParam != null)
-                            {
-                                cppClass.TemplateParameters.Add(tmplParam);
-                            }
-                            return CXChildVisitResult.CXChildVisit_Continue;
-                        }, new CXClientData((IntPtr)data));
-                        AddOrUpdateTemplateArgument(result, sourceParam, new CppTemplateArgument(sourceParam, cppClass, false));
-                        break;
-                    }
-
-                case CXTemplateArgumentKind.CXTemplateArgumentKind_Declaration:
-                    {
-                        var templateArgType = templateArg.AsDecl;
-                        var cppClass = new CppClass(GetCursorSpelling(templateArgType));
-                        AddOrUpdateTemplateArgument(result, sourceParam, new CppTemplateArgument(sourceParam, cppClass, false));
-                        break;
-                    }
-
-                case CXTemplateArgumentKind.CXTemplateArgumentKind_Integral:
-                    {
-                        AddOrUpdateTemplateArgument(result, sourceParam, new CppTemplateArgument(sourceParam, templateArg.AsIntegral));
-                        break;
-                    }
-
-                case CXTemplateArgumentKind.CXTemplateArgumentKind_Pack:
-                    {
-                        for (uint i = 0; i < templateArg.NumPackElements; i++)
-                        {
-                            var argPack = templateArg.GetPackElement(i);
-                            var packArgType = GetCppType(argPack.AsType.Declaration, argPack.AsType, cursor, data);
-                            AddOrUpdateTemplateArgument(result, sourceParam, new CppTemplateArgument(sourceParam, packArgType, argPack.AsType.TypeClass != CX_TypeClass.CX_TypeClass_TemplateTypeParm));
-                        }
-                        break;
-                    }
-
-                default:
-                    Debug.WriteLine($"[Warning]template argument with type: {templateArg.kind.ToString()} is not handled!");
-                    AddOrUpdateTemplateArgument(result, sourceParam, new CppTemplateArgument(sourceParam, templateArg.ToString()));
-                    break;
-            }
+        private void ParseTemplateArgument(Dictionary<CppType, List<CppTemplateArgument>> result, CppType sourceParam, CX_TemplateArgument templateArg, CXCursor cursor, void * data)
```

Comment:
It is very difficult for me to parse the diff of this function but I assume it was all formatting changes.

Reply:
You can enable the ignore whitespace checkbox to make these changes easier to parse.